database/dbService.js

```diff
@@ -6,9 +6,14 @@ var connection;
 module.exports = {
     /* Inserts a given user model given by Google OAuth */
     insertUser: function(user, callback) {
+
+        var placesLived = user._json.placesLived; // Handle situation in which Google gives us back an undefined user location
+        if (!placesLived) { placesLived = null; }
+        else { placesLived = placesLived[0].value; }
+
```

Comment:
!placesLived checks for undefined and null.

If it's neither of these values, pull out the first location and insert that into our DB. Otherwise, we insert a NULL into the DB.

Reply:
Lemme know if you think there's a cleaner way to do this. I handle this within the insertUser db call...cuz I don't want every parent caller to handle a possibly null location

Reply:
your logic lgtm.

just to clarify: a `user` could contain a list of places the user has lived, but we only want to save 1?

in that case something like this might be cleaner:

```
let placesLived = user._json.placesLived; // array of places user has lived OR undefined
let placeLived = placesLived ? placesLived[0].value : null; // if have places lived info, save 1 place, otherwise use null for inserting into the DB
```

then use `placeLived` in your sql query

Reply:
Google might give us back a handful of locations so we arbitrarily pick the first one. The purpose is that we show this first location in our workflow UI and the user can change it if she/he wishes. We just want to show the data some relevant data.

As for the logic, yeah I can try that. I already tried a ternary operator but didn't have much luck -- something about "?" not covering all truthies like if it was undefined or null or whatever. I'll give it another go then. @trevorhalvorson

Reply:
Update: the ternary operator "?" only checks for TRUE or FALSE. It does not check against UNDEFINED, because undefined is neither true or false. Same with null.

Explanations:
https://stackoverflow.com/questions/19277458/why-does-undefined-equals-false-return-false
https://stackoverflow.com/questions/27632391/why-null-false-and-null-true-both-return-false

Lemme know if this is good @trevorhalvorson